### PR TITLE
PostgreSQL 13.0

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

[PostgreSQL 13 has been released](https://www.postgresql.org/about/news/postgresql-13-released-2077/), so we should package it in Homebrew! I've copied the existing `postgresql.rb` formula to `postgresql@12.rb`, and made the necessary modifications for version 13.

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Could someone who knows Homebrew & PostgreSQL better than I do help me resolve the errors on CI? This appears to be the traceback:
```
Missing libraries:
  unexpected (/usr/local/lib/libecpg.6.dylib)
  unexpected (/usr/local/lib/libpgtypes.3.dylib)
  unexpected (/usr/local/lib/libpq.5.dylib)
Unexpected missing library linkage detected
postgresql@12:
  * postgresql@12 has broken dynamic library links:
```

**Comment by ruurd** [comment]
Ehm why is it referring to postgresql@12 and not postgresql@13?

**Comment by SMillerDev** [comment]
Because both exist after this pull request.

**Comment by ruurd** [comment]
But it's not something that causes this, right? For good measure I tried to check where those three libraries are on my system but I can find them at the indicated spot on my filesystem that has postgresql@12 installed.

**Comment by singingwolfboy** [comment]
@ruurd I have no idea what's causing the failure, that's why I'm asking for assistance. 😄

Based on searching Git history, it looks like @Bo98 was the one to upgrade this formula from version 11.5 to version 12.1 in e077d1877ccb31d323bac335f5f17997a558f142. Maybe he could help me figure out what needs to happen to upgrade from 12.4 to 13.0?

**Comment by fundon** [comment]
https://github.com/petere/homebrew-postgresql/blob/master/postgresql%4013.rb

**Comment by singingwolfboy** [comment]
@fundon That's a useful reference, but it doesn't explain why the automated builds are failing in this repository.

**Comment by ruurd** [comment]
> Based on searching Git history, it looks like @Bo98 was the one to upgrade this formula from version 11.5 to version 12.1 in [e077d18](https://github.com/Homebrew/homebrew-core/commit/e077d1877ccb31d323bac335f5f17997a558f142). 

If I look at that commit I wonder why prefix is referring to ${prefix} while all other paths are using ${HOMEBREW_PREFIX}`...

The other thing I wonder about is why the version regexp changed

**Comment by gromgit** [comment]
> If I look at that commit I wonder why prefix is referring to ${prefix} while all other paths are using ${HOMEBREW_PREFIX}`...

See https://docs.brew.sh/Formula-Cookbook#variables-for-directory-locations for a detailed description, but in a nutshell: `#{prefix}` (_not_ `${prefix}`) is a _versioned_ path, so you can't use it anywhere you want version independence (e.g. data dirs and the include/lib dirs to look for dependencies).

**Comment by ruurd** [comment]
> ...but in a nutshell: `#{prefix}` is a _versioned_ path, so you can't use it anywhere you want version independence (e.g. data dirs and the include/lib dirs to look for dependencies).

OK so why isn't the postgres@12 recipe not changed to use #{prefix} instead of $HOMEBREW_PREFIX in a number of places. Please have a look at postgresql@11 - you will see $HOMEBREW_PREFIX only in a small number of instances while at numerous others it uses the versioned_path variants.

@singingwolfboy can you adapt postgresql@12 from postgresql@11 instead of copypasting the existing postgresql recipe? Or let me have access to your fork please?

**Comment by ruurd** [comment]
Yay! All checks passed! @singingwolfboy you did good! You fixed it!

File: Formula/postgresql.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  sha256 "bee93fbe2c32f59419cb162bcc0145c58da9a8644ee154a30b9a5ce47de606cc"
+  url "https://ftp.postgresql.org/pub/source/v13.0/postgresql-13.0.tar.bz2"
+  sha256 "80e750be8d436b54197636a02636f8fd3263ba6779bf865b04832495ea592296"
   license "PostgreSQL"
   head "https://github.com/postgres/postgres.git"
 
   livecheck do
     url "https://www.postgresql.org/docs/current/static/release.html"
-    regex(/Release v?(\d+(?:\.\d+)+)/i)
+    regex(/Release v?(\d+(?:\.\d+)*)/i)
```
Comment (SMillerDev) [inline Formula/postgresql.rb:11]: @samford or @nandahkrishna, comments?

Comment (nandahkrishna) [reply Formula/postgresql.rb:11]: They have releases such as `12`, `12.1`, `12.2`, etc. and now, `13`. So I think we should be using `*` rather than `+`.

**Review by SMillerDev — DISMISSED** [review]
Thanks @singingwolfboy  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

File: Formula/postgresql.rb (hunk 1 above)
Comment (ruurd) [reply Formula/postgresql.rb:11]: Ah. Yes. Ofcourse.

**Comment by jonchang** [comment]
Merged in 21703acf690a1fa362ab298312ff7bcfdfd54668

**Comment by nertzy** [comment]
Thanks for this! I found a couple new issues, but it's great to make progress. 🎉 

https://github.com/Homebrew/homebrew-core/issues/62930
https://github.com/Homebrew/homebrew-core/issues/62929

**Pull request closed without merging**